SPECS/tdnf/tdnf.spec

```
@@ -99,51 +116,67 @@ systemd services for periodic automatic update
 %autosetup -p1
 
 %build
-mkdir build && cd build
-cmake \
--DCMAKE_BUILD_TYPE=Debug \
--DCMAKE_INSTALL_PREFIX=%{_prefix} \
--DCMAKE_INSTALL_LIBDIR:PATH=lib \
-..
-make %{?_smp_mflags} && make python
+%cmake \
+    -DCMAKE_BUILD_TYPE=Debug \
+    -DBUILD_SHARED_LIBS=OFF \
+    -DCMAKE_INSTALL_LIBDIR:PATH=%{_libdir} \
+    -DHISTORY_DB_DIR=%{_tdnf_history_db_dir}
+
+%cmake_build
+
+cd %{__cmake_builddir}
+%make_build python
 
 %check
 pip3 install pytest requests pyOpenSSL
 cd build && make %{?_smp_mflags} check
 
 %install
-cd build && %make_install
+%cmake_install
 find %{buildroot} -name '*.a' -delete -print
 mkdir -p %{buildroot}%{_var}/cache/tdnf
+mkdir -p %{buildroot}%{_tdnf_history_db_dir}
 ln -sf %{_bindir}/tdnf %{buildroot}%{_bindir}/tyum
 ln -sf %{_bindir}/tdnf %{buildroot}%{_bindir}/yum
+ln -sf %{_bindir}/tdnf %{buildroot}%{_bindir}/tdnfj
 install -v -D -m 0755 %{SOURCE1} %{buildroot}%{_bindir}/tdnf-cache-updateinfo
 install -v -D -m 0644 %{SOURCE2} %{buildroot}%{_libdir}/systemd/system/tdnf-cache-updateinfo.service
 install -v -D -m 0644 %{SOURCE3} %{buildroot}%{_libdir}/systemd/system/tdnf-cache-updateinfo.timer
 install -v -D -m 0644 %{SOURCE4} %{buildroot}%{_sysconfdir}/tdnf/pluginconf.d/tdnfrepogpgcheck.conf
-mv %{buildroot}%{_libdir}/pkgconfig/tdnfcli.pc %{buildroot}%{_libdir}/pkgconfig/tdnf-cli-libs.pc
-mkdir -p %{buildroot}/%{_tdnfpluginsdir}/tdnfrepogpgcheck
-mv %{buildroot}/%{_tdnfpluginsdir}/libtdnfrepogpgcheck.so %{buildroot}/%{_tdnfpluginsdir}/tdnfrepogpgcheck/libtdnfrepogpgcheck.so
+rm -f %{buildroot}%{_bindir}/jsondumptest
+rm -rf %{buildroot}%{_datadir}/tdnf
 
-pushd python
-python3 setup.py install --skip-build --prefix=%{_prefix} --root=%{buildroot}
-popd
+mkdir -p %{buildroot}%{_sysconfdir}/%{name}/protected.d
 
+pushd %{__cmake_builddir}/python
+%py3_install
+popd
 find %{buildroot} -name '*.pyc' -delete
 
-%ldconfig_scriptlets
+%post
+/sbin/ldconfig
+# Required step to ensure new history util from version 3.4.1 does not fail
+[ -f %{_tdnf_history_db_dir}/history.db ] || %{_libdir}/tdnf/tdnf-history-util init
```

To be on the safe side, let's use `if`. With the current implementation running the post-install script with a setting like `set -e`, would actually error out if `%{_tdnf_history_db_dir}/history.db` is missing. I don't know, how the scripts are ran right now, but this should be more stable in case of any behaviour changes in the future:
```suggestion
if [ -f %{_tdnf_history_db_dir}/history.db ]; then
    %{_libdir}/tdnf/tdnf-history-util init
fi
```

---

I changed this up to use an if statement as I agree its a bit more clear than using bash `||`. The goal of it in the first place is to only run the `tdnf-history-util init` if the history.db file is not in place already, given it makes all tdnf calls fail if it is not.

---

Huh, not sure why I though that it's going to bail out on `||` then. Sorry for the confusion.